### PR TITLE
Support http proxy

### PR DESCRIPTION
**Pull request by egut**
This patch enables users to add a https proxy agent if they want/need to.

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -69,9 +69,12 @@ client.prototype.request = function(method, path, params, callback){
         hostname: this.api_host,
         port: 443,
         method: method.toUpperCase(),
-        path: path,
+        path: path
     };
 
+    if(this.proxy_agent)
+        http_options["agent"] = this.proxy_agent;
```
Comment (brettlangdon) [inline lib/client.js]: can you please update to add brackets for this if statement to keep consistency with the rest of the code.

```js
if(this.proxy_agent){
    http_options["agent"] = this.proxy_agent;
}
```

**Comment by brettlangdon** [comment]
also, can you please update the docstring in `lib/index.js` for `initialize` to include `proxy_agent`.

**Comment by brettlangdon** [comment]
thanks for the PR! feel free to add yourself to the list of contributors in `package.json` if you want.

**Comment by egut** [comment]
Thanks for the review, I think I have fixed all of them.

**Pull request merged**